slpv2.rflx

```diff
@@ -0,0 +1,289 @@
+      if Function_ID = FI_SERVICE_REPLY;
+
+   -- 8.3. Service Registration
+   type Service_Registration is
+      message
+         URL_Entry : URL_Entry;
+         Length_Of_Service_Type_String : U16;
+         Service_Type : Opaque with Size => Length_Of_Service_Type_String * 8;
+         Length_Of_Scope_List : U16;
+         Scope_List : Opaque with Size => Length_Of_Scope_List * 8;
+         Length_Of_Attribute_List_String : U16;
+         Attribute_List : Opaque with Size => Length_Of_Attribute_List_String * 8;
+         Number_Of_Attribute_Authentication_Blocks : U8;
+         Attribute_Authentication_Blocks : Authentication_Blocks
+            with Size => Number_Of_Attribute_Authentication_Blocks * 8;
+      end message;
+   for SLPv2 use (Payload => Service_Registration)
+      if Function_ID = FI_SERVICE_REGISTRATION;
+
+   -- 8.4. Service Acknowledgement
+   type Service_Acknowledgement is
+      message
+         Error_Code : Error;
+      end message;
+   for SLPv2 use (Payload => Service_Acknowledgement)
+      if Function_ID = FI_SERVICE_ACKNOWLEDGE;
+
+   -- 8.5. Directory Agent Advertisement
+   type Directory_Agent_Advertisement is
+      message
+         Error_Code : Error;
+         DA_Stateless_Boot_Timestamp : U32;
+         Length_Of_URL : U16;
+         URL : Opaque with Size => Length_Of_URL * 8;
+         Length_Of_Scope_List : U16;
+         Scope_List : Opaque with Size => Length_Of_Scope_List * 8;
+         Length_Of_Attribute_List_String : U16;
+         Attribute_List : Opaque with Size => Length_Of_Attribute_List_String * 8;
+         SLP_SPI_String_Length : U16;
+         SLP_SPI_String : Opaque with Size => SLP_SPI_String_Length * 8;
+         Number_Of_Authentication_Blocks : U8;
+         Authentication_Blocks : Authentication_Blocks
+            with Size => Number_Of_Authentication_Blocks * 8;
+      end message;
+   for SLPv2 use (Payload => Directory_Agent_Advertisement)
+      if Function_ID = FI_DA_ADVERTISEMENT;
+
+   -- 8.6. Service Agent Advertisement
+   type Service_Agent_Advertisement is
+      message
+         Length_Of_URL : U16;
+         URL : Opaque with Size => Length_Of_URL * 8;
+         Length_Of_Scope_List : U16;
+         Scope_List : Opaque with Size => Length_Of_Scope_List * 8;
+         Length_Of_Attribute_List_String : U16;
+         Attribute_List : Opaque with Size => Length_Of_Attribute_List_String * 8;
+         Number_Of_Authentication_Blocks : U8;
+         Authentication_Blocks : Authentication_Blocks
+            with Size => Number_Of_Authentication_Blocks * 8;
+      end message;
+   for SLPv2 use (Payload => Service_Agent_Advertisement)
+      if Function_ID = FI_SA_ADVERTISEMENT;
+
+   -- 9.1. Service Location Protocol Extensions
+   -- TODO: Support verification of extensions; not possible/feasible with RecordFlux 5.0, see issue comment https://github.com/Componolit/RecordFlux-specifications/issues/20#issuecomment-752084366
```

Comment:
I have created an issue to track that limitation. Please replace the comment by a reference to the issue:

```
-- ISSUE: Componolit/RecordFlux#598
```

Reply:
Fixed.